### ISSUE TITLE
[Python] regex comments

### ISSUE THREAD

**Issue by buildersbrewery** [sublimehq/Packages#1971]
<img width="653" alt="image" src="https://user-images.githubusercontent.com/13032004/57740860-2cda2800-76ba-11e9-8b37-a1fcdd7ebf2f.png">

```python
    regex = r'''(?xi)                   # ignore case and whitespace
        (?:
            # ^                                               # BOL
            # (?P<path>..[^:]*)                               # path
            # \:\:                                            # two colons
            (?:(?P<warning>WARN)|(?P<error>ERROR))          # WARN or ERROR
            \:\:                                            # two colons
            \s                                              # whitespace
            \(                                              # left brace
            \s*                                             # optional whitespace
            (?P<line>[0-9]+)                                # named group: line
            ,                                               # comma
            \s*                                             # optional whitespace
            (?P<col>[0-9]+)                                 # named group: col
            \)                                              # right brace
            \:                                              # colon
            \s                                              # whitespace
            (?P<message>.*)                                 # named group: message
            $                                               # EOL
        )
    '''
```

**Comment by FichteFoll** [comment]
Comments in python's regex syntax are a rough best effort in that they only match when no other significant characters are following, as opposed to tracking the presence of a `(?x) ` Switch, because extended mode can also be enabled with a flag in an argument.

**Comment by deathaxe** [comment]
Fixed by #3098

![grafik](https://user-images.githubusercontent.com/16542113/171017690-662947a6-cfe5-4ab1-88e1-5be86f9f3b08.png)
